### ISSUE TITLE
Questions about CNTK 204 Tutorial

### ISSUE THREAD

**Issue by kkc0923** [Microsoft/CNTK#3375]
Hi. I have a question about CNTK 204: Sequence to Sequence Networks with Text Data.
https://cntk.ai/pythondocs/CNTK_204_Sequence_To_Sequence.html

1. Why it uses true labels as its history for training time? Is there any reason that its real output can not be used as history (like evaluation time)?

2. It seems that labels in dataset has already sentence start tag. But in the tutorial, it removes start tag in create_criterion_function (by using C.sequence.slice) and adds start tag again in create_model_train (by using C.layers.Delay). I think this process is rebundant but CNTK 204 do this. Why this step is necessary?

Thanks.

**Comment by VHellendoorn** [comment]
Hi, I can't speak for the team, but here are some thoughts:
1. If you used the emitted symbols, the supervisory signal might become useless or worse after a few words. For instance, if the decoder comes up with a different but reasonable translation, its error for most words will be maximal even though its translation might be fine. Training like that is not impossible, but it does take a more complex kind of training setup. By feeding it the ground-truth, it can correct early on, so using the true labels makes training a lot easier because it splits the task into simpler prediction tasks. The model can still learn good translation this way.
2. The slicing and delay is so that the source and target data are on the same dynamic sequence axis, which CNTK keeps track off. I am pretty sure you can also do it by slicing both sequences (removing the last word from source and the first word from target), but perhaps this was easier for the eventual greedy decoder, or just a good way to demonstrate different functions :)

**Comment by kkc0923** [comment]
@VHellendoorn Thank you for answer!
